src/composition-engine.js

```diff
@@ -69,6 +69,37 @@ function tryActivateViewModel(context) {
   return context.viewModel.activate(context.model) || Promise.resolve();
 }
 
+class SwapStrategies {
+  // animate the next view in before removing the current view;
+  before(viewSlot, previousView, callback) {
+    let promise = Promise.resolve(callback());
```

Comment:
Is this public API surface? What is the contract exactly?

My understanding is that in the PR code those methods are used internally only and never escape this module.
In that case, `callback()` is guaranteed to return a `Promise` already...